osquery/tables/system/magic.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <stdio.h>
+#include <magic.h>
+
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+QueryData genMagicData(QueryContext& context) {
+  QueryData results;
+  magic_t magic_cookie;
```

Comment:
Like @marpaia pointed out: this is actually an opaque pointer type:
```cpp
typedef struct magic_set *magic_t;
```

If this is initialized: `magic_t magic_cookie = nullptr;` we'll benefit from static analysis.